Add creation of VRG roles using manifest work

ShyamsundarR opened the pull request:
On the managed cluster, the klusterlet service requires rights for CRUD operations on VRG Kind.
    
To enable the same, create the required ClusterRoles and ClusterRoleBindings as manifest work for the knusterlet service.
    
Signed-off-by: Shyamsundar Ranganathan <srangana@redhat.com>

ShyamsundarR commented:
Only look at the second [commit](https://github.com/RamenDR/ramen/pull/23/commits/e160939e6c01baa62295e5ddaf926798038288ed)

The first commit is from PR #22 and is added here to provide continuity.

BenamarMk commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -189,9 +202,15 @@ func (r *ApplicationVolumeReplicationReconciler) createVRGManifestWork(
+	if err := r.createOrUpdateVRGManifestWork(
 		subscription.Name, subscription.Namespace, homeCluster); err != nil {
-		r.Log.Error(err, "failed to create or update manifest")
+		r.Log.Error(err, "Failed to create or update VolumeReplicationGroup manifest")
```
error log message should start in lowercase per: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

BenamarMk reviewed (approved):
@ShyamsundarR I have a minor comment only. We can merge it so that I can fix the merge conflicts that will arise from your PR and mine.  I think we should collaborate better so that we don't work on the same thing (especially now that we are constantly refactoring as things change) and end up spending too much time on merge hell. I think Jira is a good place to start adding tasks so that we know what we are changing.

ShyamsundarR commented:
> @ShyamsundarR I have a minor comment only. We can merge it so that I can fix the merge conflicts that will arise from your PR and mine. I think we should collaborate better so that we don't work on the same thing (especially now that we are constantly refactoring as things change) and end up spending too much time on merge hell. I think Jira is a good place to start adding tasks so that we know what we are changing.

Addressed review comments (separate commit for easier review, so rebase merge is better in this instance).

I did expect a conflict, but was looking at VRG creation at managed cluster, hence added the additional roles. Also marked it WIP as a result, as I was not sure if this was being handled.

We can add tasks as issues in here (easier visibility for anyone looking at the project).

ShyamsundarR commented on controllers/applicationvolumereplication_controller.go:
```diff
@@ -295,39 +303,99 @@ func (r *ApplicationVolumeReplicationReconciler) extractHomeClusterAndPeerCluste
-	homeCluster string) error {
-	r.Log.Info("attempt to create and update ManifestWork")
+func (r *ApplicationVolumeReplicationReconciler) createOrUpdateVRGRolesManifestWork(namespace string) error {
+	// TODO: Enhance to remember clusters where this has been checked to reduce repeated Gets of the object
```
If we decide to merge this, need to track the above in an issue and also need to ensure this is the right manner of delivering the `ClusterRoles` and `ClusterRoleBindings`, rather than as a policy(?) to all `ManagedClusters`.

BenamarMk replied:
I think we can merge it and create an issue to track it. I think it is fine and I only have one issue doing it this way, which is, how do we ensure that the *rolebindings is delivered before the VRG? I think we can always create a VRG only after we make sure that the state of the *RoleBindings MW is `Applied`.

ShyamsundarR replied:
Klusterlet fails to create VRG and retries the reconciliation of manifestWork, and succeeds once the roles are updated. So order is not mandatory (yet?) in this case.

I did think of pushing Roles to all MCs as some form of init routine, but not all MCs would be in a DR relationship and hence do not need these roles. So there are other alternatives than just letting the system reach healthy state eventually.
